yt/data_objects/selection_objects/spheroids.py

```diff
@@ -57,12 +57,6 @@ def __init__(
         super().__init__(center, ds, field_parameters, data_source)
         # Unpack the radius, if necessary
         radius = fix_length(radius, self.ds)
-        if radius < self.index.get_smallest_dx():
```

Comment:
isn't *this* where we should be adding a `self.ds.index` for backward compat ?

Reply:
I agree with @cphyc. Taking this out makes these objects significantly lighter.

Reply:
To me the issue is that we have some public API that *requires* the index to be built-- `ds.field_info` being one of them ­-- yet they don't all trigger the creation of the index.

Reply:
I think these tests are not reflecting real usage. One would never really access the field info container in this way. That said, the brittleness of `field_info` needing other actions before it can be access has always bugged me. PR #5250 is a potential fix for this that would allow us to remove the `ds.index` calls in these tests.

Reply:
No guarantees, but it's hard to imagine. That this previously worked was a side-effect of suboptimal functionality. This is a potentially huge performance improvement. I'm in favor of merging this whether or not it breaks things downstream.